whylog/assistant/regex_assistant/regex_match.py

```diff
@@ -44,11 +45,18 @@ def update_by_regex(self, new_regex):
         Updates self.param_groups so that they correspond to new_regex groups
         """
 
-        groups = regex_groups(new_regex, self.line_text)
-
         if new_regex[-1] != '$':
             new_regex += '$'
 
+        try:
+            verify_regex(new_regex, self.line_text)
+        except NotMatchingRegexError:
+            self.regex = new_regex
```

Comment:
We talked with Paweł many times that user should have a possibility to update not matching regex. Here you catch an error and show information about an error to the front.